cpp/examples/sbml_to_memilio.cpp

```diff
@@ -0,0 +1,921 @@
+            << std::endl;
+    example << "sir.print_table({" << table << "}, 10, 3, file);" << std::endl;
+    example << "file.close();" << std::endl;
+
+    example << "}";
+    example.close();
+
+    return true;
+}
+
+/*
+    @brief Create the additional lines for the CmakeLists.txt file
+
+    @param filename The name of the file to be processed
+
+    Extracts the filename using :cpp:func:`get_filename(const char* filename)`. Then it writes the generic compilation commands to a file called "CMakeListsAddition.txt" that is stored in the current working directory.
+*/
+bool modify_cmakelists(const char* filename)
+{
+    std::string lowercase_name = boost::to_lower_copy<std::string>(get_filename(filename));
+    std::ofstream modifications;
+    modifications.open("CMakeListsAddition.txt", std::ios::app);
+    modifications << "add_executable(ex_" << lowercase_name << " " << lowercase_name << ".cpp)" << std::endl;
+    modifications << "target_link_libraries(ex_" << lowercase_name << " PRIVATE memilio " << lowercase_name << ")"
+                  << std::endl;
+    modifications << "target_compile_options(ex_" << lowercase_name
+                  << " PRIVATE ${MEMILIO_CXX_FLAGS_ENABLE_WARNING_ERRORS})" << std::endl;
+    modifications.close();
+    modifications.open("CMakeListsFolderNames.txt", std::ios::app);
+    modifications << "add_subdirectory(models/" << lowercase_name << ")" << std::endl;
+    modifications.close();
+    return true;
+}
+
+/* 
+    @brief Calls clang-format to format the generated files
+
+    @param filename The name of the file to be processed
+
+    Calls clang-format using the `system` command. The command is generated using the filename.
+*/
+int format_files(const char* filename)
+{
+    std::string lowercase_name = boost::to_lower_copy<std::string>(get_filename(filename));
+    std::string command =
+        std::string("clang-format -i --files= ") + lowercase_name + ".cpp " + lowercase_name + "/*.[hc]*";
+    return system(command.c_str());
+}
+
+/* 
+    @brief Runs the program
+    @param argc The number of command line arguments
+    @param argv The command line arguments, here the name of the file that should be processed
+
+    This function expects exaclty one parameter which should be the path to the file that should be processed. If more or less parameters are given, it raises an error. If the correct number of parameters is given, it treats the file as a SBML file and generates the MEmilio code for it.
+*/
+int main(int argc, char* argv[])
+{
+    if (argc != 2) {
+        std::cout << "Please provide a SBML file at startup!" << std::endl << std::endl;
+        return 1;
+    }
+    const char* filename = argv[1];
+    SBMLDocument* document;
+    SBMLReader reader;
+
+    document = reader.readSBML(filename);
+
+    auto model = document->getModel();
+
+    if (!verify_model_suitability(model)) {
+        return 1;
```

Comment:
Most commonly used exit codes start at 127, so if you like, you could use incrementing exit codes so it is easier to see which function failed.